pkg/logging/recorded.go

```diff
@@ -20,44 +20,61 @@ func WithNoRecord(logger logr.Logger) logr.Logger {
 	return logger.WithValues("eventType", EventTypeDontRecord)
 }
 
-var _ logr.Logger = &EventRecordingLogger{}
+var _ logr.LogSink = &EventRecordingLogSink{}
 
-func WithEventRecorder(logger logr.Logger, recorder record.EventRecorder, object runtime.Object) logr.Logger {
-	return &EventRecordingLogger{
-		Logger:   logger,
-		recorder: recorder,
-		object:   object,
-		values:   []interface{}{},
-	}
-}
-
-type EventRecordingLogger struct {
-	logr.Logger
+type EventRecordingLogSink struct {
+	logger   logr.LogSink
+	info     logr.RuntimeInfo
 	recorder record.EventRecorder
 	object   runtime.Object
 	values   []interface{}
 }
 
-func (erl *EventRecordingLogger) WithValues(values ...interface{}) logr.Logger {
-	return &EventRecordingLogger{
-		Logger:   erl.Logger.WithValues(values...),
+// Init implements logr.LogSink.
```

Comment:
Not too sure what the convention is to be honest!

I decided to put those comments in because I thought they'd make it clear that these `EventRecordingLogSink` methods are needed to allow `EventRecordingLogSink` to implement `logr.LogSink`'s interface. Especially given that a some of these methods aren't used in the rest of our codebase.